tests/integration/test_validators.py

```diff
@@ -195,6 +201,54 @@ def test_string_uuid(self, value):
 
         assert result is None
 
+    def test_ref(self):
+        schema = {
+            "$ref": "#/$defs/Pet",
+            "$defs": {
+                "Pet": {
+                    "required": ["id", "name"],
+                    "properties": {
+                        "id": {"type": "integer", "format": "int64"},
+                        "name": {"type": "string"},
+                        "tag": {"type": "string"},
+                    },
+                }
+            },
+        }
+        validator = OAS30Validator(schema, format_checker=oas30_format_checker)
+
+        result = validator.validate({"id": 1, "name": "John"})
+        assert result is None
+
+        with pytest.raises(ValidationError) as excinfo:
+            validator.validate({"name": "John"})
+
+        error = "'id' is a required property"
+        assert error in str(excinfo.value)
+
+    @pytest.mark.xfail(reason="Issue #20")
+    def test_ref_nullable(self):
+        schema = {
+            "nullable": True,
+            "allOf": [
+                {
+                    "$ref": "#/$defs/Pet",
```

Comment:
This subschema is invalid (not nullable). That means the whole allof should be invalid isn't it?

Reply:
Is it? I'd have expected the reference this to flatten to:

```json
{
    "nullable": true,
    "allOf": [
        {
            "required": ["id", "name"],
            "properties": {
                "id": {"type": "integer", "format": "int64"},
                "name": {"type": "string"},
                "tag": {"type": "string"}
            }
        }
    ]
}
```

which would in-turn flatten to:

```json
{
    "nullable": true,
    "required": ["id", "name"],
    "properties": {
        "id": {"type": "integer", "format": "int64"},
        "name": {"type": "string"},
        "tag": {"type": "string"}
    }
}
```

Is my understanding incorrect? It might well be! :smile:

Reply:
Ahh, wait, I see your top-level comment now.

> > [allOf](https://json-schema.org/understanding-json-schema/reference/combining.html#allof) can not be used to “extend” a schema to add more details to it in the sense of object-oriented inheritance. Instances must independently be valid against “all of” the schemas in the allOf. See the section on [Extending Closed Schemas](https://json-schema.org/understanding-json-schema/reference/object.html#extending) for more information.
> 
> [json-schema.org/understanding-json-schema/reference/combining.html#allof](https://json-schema.org/understanding-json-schema/reference/combining.html#allof)

So this being the case, I wonder if the schemas should be outright rejected as invalid?